api/src/main/java/com/datatorrent/api/StatsListener.java

```diff
@@ -115,6 +115,30 @@
     List<OperatorResponse> getOperatorResponse();
   }
 
+  /**
+   * An interface to the DAG. Stats listener can get information about
+   * operator or other elements in the DAG through this interface. currerntly
+   * we only provide method to extract the operator name based on the physical
+   * id of the operator. In future more methods can be added.
+   *
+   */
+  interface StatsListenerContext
+  {
+    String getOperatorName(int id);
+
+    DAG createDAG();
+  }
+
+  /**
+   * If StatsListener implements ContextAwareStatsListener interface, then engine will
+   * provide a reference to StatsListenerContext using which listener can examine
+   * current state of the DAG.
+   */
+  interface ContextAwareStatsListener
```

Comment:
Extending stats listener will break API compatibility as one more additional method needs to be implemented for existing listener that user have written.